src/app/shared/services/auth.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Router } from '@angular/router';
+import { throwError,BehaviorSubject } from 'rxjs';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from "@angular/core";
+import { catchError, tap } from 'rxjs/operators';
+import { User } from '../models/auth.model';
+
+
+export interface AuthResponseData {
+  idToken: string,
+  email: string,
+  refreshToken: string,
+  expiresIn: string,
+  localId: string,
+  registered?: boolean,
+}
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  user = new BehaviorSubject<User>(null!);
+  private tokenExpirationTimer: any;
+
+  constructor(private http: HttpClient, private router : Router) {}
+
+  autoLogin() {
+    const userData: {
```

Comment:
Also token info can be stored in HTTP only cookies, so browser will automatically send it to server, and for example you can add endpoint on backend to get token info from cookie and then return user info, just for your knowledge